File: src/Java/adventofcode2017/Day01.java

package adventofcode2017;

import java.util.stream.IntStream;

import utils.Input;

/**
 * https://adventofcode.com/2017/day/1
 */
public class Day01 {

    static int calc(String input, int dist) {
        int[] chars = input.chars().map(i -> i - '0').toArray();
        int n = chars.length;
        return IntStream.range(0, n)
                .map(i -> chars[i] == chars[(i+dist)%n] ? chars[i] : 0)
                .sum();
    }

    public static void main(String[] args) throws Exception {
        String input = new Input(2017, 1).strings().get(0);
        System.out.println(calc(input, 1));
        System.out.println(calc(input, input.length() / 2));
    }

}

File: src/Python/2020/Day01.py

from utils import Input, printResult

# https://adventofcode.com/2020/day/1

input = list(map(int, Input(2020, 1).lines()))

for x in range(len(input)):
    for y in range(x+1, len(input)):
        a, b = input[x], input[y]
        if a + b == 2020:
            part1 = a * b
        elif a + b < 2020 and (2020 - a - b) in input[y+1:]:
            part2 = a * b * (2020 - a - b)

printResult(1, part1)
printResult(2, part2)
